homeassistant/components/matter/sensor.py

```diff
@@ -0,0 +1,167 @@
+"""Matter sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import partial
+from typing import TYPE_CHECKING, Any
+
+from chip.clusters import Objects as clusters
+from chip.clusters.Types import Nullable, NullValue
+from matter_server.common.models import device_types
+from matter_server.common.models.device_type_instance import MatterDeviceTypeInstance
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    LIGHT_LUX,
+    PERCENTAGE,
+    VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+    Platform,
+    UnitOfPressure,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import MatterEntity, MatterEntityDescriptionBaseClass
+
+if TYPE_CHECKING:
+    from .adapter import MatterAdapter
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter sensors from Config Entry."""
+    matter: MatterAdapter = hass.data[DOMAIN][config_entry.entry_id]
+    matter.register_platform_handler(Platform.SENSOR, async_add_entities)
+
+
+class MatterSensor(MatterEntity, SensorEntity):
+    """Representation of a Matter sensor."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    entity_description: MatterSensorEntityDescription
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        measurement: Nullable | float | None
+        measurement = _get_attribute_value(
+            self._device_type_instance,
+            # We always subscribe to a single value
+            self.entity_description.subscribe_attributes[0],
+        )
+
+        if measurement is NullValue or measurement is None:
+            measurement = None
+        else:
+            measurement = self.entity_description.measurement_to_ha(measurement)
+
+        self._attr_native_value = measurement
+
+
+def _get_attribute_value(
+    device_type_instance: MatterDeviceTypeInstance,
+    attribute: clusters.ClusterAttributeDescriptor,
+) -> Any:
+    """Return the value of an attribute."""
+    # Find the cluster for this attribute. We don't have a lookup table yet.
```

Comment:
why is this needed as we have direct access to the cluster's attributes (and values) ?

Reply:
Yeah, I wasn't sure if we had a better way now. I didn't change it. Please make a PR to clean it up.

Reply:
Yeah, we can now access each attribute by key which is much faster. I'll do a follow-up PR for that later. It's fine now as is.